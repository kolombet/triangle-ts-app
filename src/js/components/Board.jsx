import React, { PureComponent } from "react";

export default class Board extends PureComponent {
  render() {
    return (
      <div data-ts="Board">
        <div data-ts="Panel">{this.props.children}</div>
      </div>
    );
  }
}
