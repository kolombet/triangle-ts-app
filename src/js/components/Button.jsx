import React, { PureComponent } from "react";
import PropTypes from "prop-types";

export default class Button extends PureComponent { 
  static propTypes = {
    title: PropTypes.string.isRequired,
    onClick: PropTypes.func,
    disabled: PropTypes.bool
  }

  render() {
    return (
      <button
        disabled={this.props.disabled ? "disabled" : ""}
        data-ts="Button"
        type="submit"
        className="ts-primary"
        onClick={this.props.onClick}
      >
        {this.props.title}
      </button>
    );
  }
}
