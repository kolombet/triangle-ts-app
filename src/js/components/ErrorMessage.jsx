import React, { PureComponent } from "react";

export default class ErrorMessage extends PureComponent {
  render() {
    return (
      <dl className="ts-errors">
        <dt>{this.props.title}</dt>
        {this.props.description && <dd>{this.props.description}</dd>}
      </dl>
    );
  }
}
