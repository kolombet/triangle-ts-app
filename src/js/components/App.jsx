import React from "react";
import TrianglesList from "./TrianglesList.jsx";
import Form from "./Form.jsx";
import ControlBar from "./ControlBar.jsx";
import Board from "./Board";

const App = () => (
  <div>
    <Board>
      <ControlBar />
      <TrianglesList />
    </Board>
    <Board>
      <h2>Add a new triangle</h2>
      <Form />
    </Board>
  </div>
);
export default App;
