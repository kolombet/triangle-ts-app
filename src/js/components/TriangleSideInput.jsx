import React, { PureComponent } from "react";
import ErrorMessage from "./ErrorMessage";
import PropTypes from "prop-types";

export default class TriangleSideInput extends PureComponent {
    static propTypes = {
        value: PropTypes.string,
        label: PropTypes.string,
        id: PropTypes.string,
        onValueChange: PropTypes.func
    }

    render() {
        const isError = this.props.value < 0;
        return (
            <label>
                <span>{this.props.label}</span>
                <input
                    required
                    type="number"
                    className="form-control"
                    id={this.props.id}
                    value={this.props.value}
                    onChange={this.props.onValueChange}
                />
                {isError && (
                    <ErrorMessage title="Triangle sides size must be positive" />
                )}
            </label>
        );
    }
}