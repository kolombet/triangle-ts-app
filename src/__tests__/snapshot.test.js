import React from "react";
import ReactDOM from "react-dom";
import { Provider } from "react-redux";
import { create } from "react-test-renderer";
import store from "../js/store/index";

import App from "../js/components/App";
import Button from "../js/components/Button";
import ErrorMessage from "../js/components/ErrorMessage";
import Board from "../js/components/Board";
import TriangleSideInput from "../js/components/TriangleSideInput";

describe("Button component", () => {
  test("Enabled button matches the snapshot", () => {
    const component = create(<Button title="test"/>);
    expect(component.toJSON()).toMatchSnapshot();
  });

  test("Disabled button matches the snapshot", () => {
    const component = create(<Button disabled={true} title="test"/>);
    expect(component.toJSON()).toMatchSnapshot();
  });
});

describe("Error message component", () => {
  test("Error message matches the snapshot", () => {
    const component = create(<ErrorMessage title="title" description="description"/>);
    expect(component.toJSON()).toMatchSnapshot();
  });
});

describe("Board component", () => {
  test("Board component matches the snapshot", () => {
    const component = create(<Board>test</Board>);
    expect(component.toJSON()).toMatchSnapshot();
  });
});

describe("Triangle side input component", () => {
  test("Triangle side input matches the snapshot", () => {
    const component = create(<TriangleSideInput label="Side:" id="side" val={5}/>);
    expect(component.toJSON()).toMatchSnapshot();
  });
});