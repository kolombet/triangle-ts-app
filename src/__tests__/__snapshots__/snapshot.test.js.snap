// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board component Board component matches the snapshot 1`] = `
<div
  data-ts="Board"
>
  <div
    data-ts="Panel"
  >
    test
  </div>
</div>
`;

exports[`Button component Disabled button matches the snapshot 1`] = `
<button
  className="ts-primary"
  data-ts="Button"
  disabled="disabled"
  type="submit"
>
  test
</button>
`;

exports[`Button component Enabled button matches the snapshot 1`] = `
<button
  className="ts-primary"
  data-ts="Button"
  disabled=""
  type="submit"
>
  test
</button>
`;

exports[`Error message component Error message matches the snapshot 1`] = `
<dl
  className="ts-errors"
>
  <dt>
    title
  </dt>
  <dd>
    description
  </dd>
</dl>
`;

exports[`Triangle side input component Triangle side input matches the snapshot 1`] = `
<label>
  <span>
    Side:
  </span>
  <input
    className="form-control"
    id="side"
    required={true}
    type="number"
  />
</label>
`;
